core/domain/base_model_validators.py

```diff
@@ -380,6 +382,25 @@ def validate(cls, item):
         for func in cls._get_external_instance_custom_validation_functions():
             func(item, cls.field_name_to_external_model_references)
 
+    @classmethod
+    def validate_deleted(cls, item):
+        """Validate that the models marked as deleted are hard-deleted after
+        nine weeks.
+
+        Args:
+            item: datastore_services.Model. Entity to validate.
+        """
+        cls.errors.clear()
+        date_now = datetime.datetime.utcnow()
+        date_before_which_models_should_be_deleted = (
+            date_now - PERIOD_TO_HARD_DELETE_MODELS_MARKED_AS_DELETED)
+        if item.last_updated < date_before_which_models_should_be_deleted:
+            cls._add_error(
+                'entity %s' % ERROR_CATEGORY_STALE_CHECK,
+                'Entity id %s: '
+                'model marked as deleted older more than 8 weeks' % item.id
```

Comment:
deleted older more than --> deleted is older than

Reply:
Done.